omnilib/data-final-fixes.lua

```diff
@@ -1,8 +1,10 @@
-for _,pack in pairs({"automation-science-pack","logistic-science-pack","chemical-science-pack","production-science-pack","military-science-pack","utility-science-pack"}) do
-	data.raw.tool[pack].icon = nil
-	data.raw.tool[pack].icons = {{icon="__omnilib__/graphics/icons/science-pack/"..pack..".png",icon_size=64}}
-	if data.raw.technology[pack] then
-		data.raw.technology[pack].icon = "__omnilib__/graphics/technology/"..pack..".png"
+if not mods["Krastorio2"] then
+	for _,pack in pairs({"automation-science-pack","logistic-science-pack","chemical-science-pack","production-science-pack","military-science-pack","utility-science-pack"}) do
```

Comment:
Gonna be honest, this block shouldn't be here. I understand this is Omni's eponymous project and I will respect that it will not be changed without his consent but this is causing visual issues for no good reason.

Reply:
Krastorio2 is a relatively new mod on the block, this block of code has been in existence for quite a while (about mid Factorio 0.15). While we respect the need for compatability with other mods, adding in a "don't replace if x mod exists" has the potential to be messy if we open that floodgate. I will take a further look into how much "immersion" is broken with the replacement and make a call on accepting this or not then.
A more elegant solution may exist for this, i will toy with it over the next week.

Reply:
With your feedback, I would like to come to a mutually agreeable resolution. I agree the method I have proposed here is not elegant and has the possibility of incurring a technical debt. I'm missing the context in which this was originally written, so I have a few questions to establish what the best route for a fix is:

- Is this block still necessary?
- Is the purpose it serves specific to omnilib, or does it belong in one of the omnimods?
- Is there a reason why we do this override unconditionally?
- Is there a case against making a setting to control this behavior?

The issues I see are as follows:
- The override is currently unconditional
  - This means that we're making the assumption that no other mod will want to chance science pack icons, or if they do we should have precedence over them
    - This has been the cause of a good bit of grief for myself as every mod I have tried in recent memory (bobs, K2) has had their icon preferences overwritten by omnilib
- The user has no recourse if this behavior is undesirable
  - This is not currently controlled by a setting, and the action is taken in final-fixes meaning one must write a mod that depends on omnilib or edit the source
    - For the former, I did such a thing and it was marked incompatible with omnilib within a few weeks. I was unable to get a concrete response as to what I could do.

Reply:
In addendum to the above, I feel that if this type of override is desirable we should be either have checks against what mods made this override necessary, or if the possible list for such a thing is too large, move this into a scope where it's only called when necessitated by one of the present omnimods (omniscience?).

Reply:
Ill answer the questions in order:
 - the block is at present required for a specific task and is considered non-negotiable when running with bobs mods. (reasons are petty)
 - the purpose is to revert bobs colour tech update and it runs in omnilib as it is for "all" conditions
 - pretty much as above
 - My proposal is to fix the problem by simply forcing bobs setting off and hiding it, thus removing the need to brute-force it.
 - The change associated with bobs colour update is considered "Heresy" by OmnissiahZelos, and thus had to be removed.
 - I am hoping that my proposed change, which removes the "forced icon update" to simply not need to exist.

Reply:
As you seem to understand the reasons for it, I'll leave it as your call. Thank you for the explanation.